### ISSUE TITLE
FilterMusic has false positives in documents with tables

### ISSUE THREAD

**Issue by akhudek** [tesseract-ocr/tesseract#1255]
### Environment

* **Tesseract Version**: v4 commit number ebbfc3ae8df85c351002000a76900e3086375e7b
* **Platform**: Darwin 17.3.0 Darwin Kernel Version 17.3.0: Thu Nov  9 18:09:22 PST 2017; root:xnu-4570.31.3~1/RELEASE_X86_64 x86_64

### Current Behavior:
The table header and parts of the cell contents are not recognized due to the binarized version of the document having critical parts of the table removed.
![test-table](https://user-images.githubusercontent.com/147633/34470838-5eae68cc-ef08-11e7-9a0b-fe66f195c25c.png)
![testb](https://user-images.githubusercontent.com/147633/34470846-897c6996-ef08-11e7-9deb-b726e60a87ca.png)

### Expected Behavior:
If you disable FilterMusic, it works fine.
![testb2](https://user-images.githubusercontent.com/147633/34470863-e6ca0900-ef08-11e7-8078-3b48f02ff817.png)

### Suggested Fix:
It's not clear to me what the motivation was for trying to find and identify sheet music. Is it possibly for the table detector? If so, given that the table detection is fairly poor anyways and the extraction isn't complete, I'd suggest just removing this music filter code. Alternately, maybe expose an option to disable it?

**Comment by zdenop** [comment]
Now it is possible to use `pageseg_apply_music_mask` e.g. with `-c pageseg_apply_music_mask=0`

**Comment by stweil** [comment]
I wonder whether the current default is reasonable. If most documents give better results when this parameter is `false`, then that should be the default.

**Comment by zdenop** [comment]
IMO it will need more investigations:
* regarding tables - PR 2732 seems to be still just part solution (some lines are removed in tables)
* _maybe_ current setup helps in other kind of documents - it should be tested on wider range of documents (unfortunately we do not have such test base at the moment). I have internal info from developer that use tesseract in thousands of docs and their experience is that if you "fix" page segmentation on one type of document - you break it on other types. So they suggest to be very careful when playing with page segmentation. BTW: they recognize tesseract page segmentation as best available opensouce general page segmentation library...

**Comment by bertsky** [comment]
I've just come across this through a lot of painful debugging of table-rich images. I had never heard of `pageseg_apply_music_mask` until I found it in the code – it kind of smiled back at me and said ["I'm all here!"](https://en.wikipedia.org/wiki/The_Hare_and_the_Hedgehog).

@stweil @zdenop IMO tables are much more important/prevalent than music sheets, and for the latter you'd probably need better/specialized software anyway. So I would argue the default *should* be changed. In the very least, this should be well documented and made more visible (perhaps by shipping a config file for it).

BTW, the [other problem](https://github.com/tesseract-ocr/tesseract/pull/2732#issuecomment-546703111) with line detection (there's more of course) seems to be the proximity between separators and text in low-density images – this is even harder than the music/not-music decision, because you can easily get false positive horizontal or vertical lines in normal text. (Pragmatically you can upsample such images as a workaround, therefore perhaps the code can be made more robust by using additional filters or optimise some parameters...)

Here's an example that also shows how naive the music sheet detection is:
![vhlinefinding_img](https://user-images.githubusercontent.com/38561704/134547580-e6b50fdf-d57d-453e-9ffe-7f1c7829a254.png)

**Comment by bertsky** [comment]
> So I would argue the default _should_ be changed. In the very least, this should be well documented and made more visible (perhaps by shipping a config file for it).

Also, I don't think simply _suppressing_ the staves is the correct thing to do. They should (also) be polygonalized and used as partitions during layout analysis, to be presented as (special) regions (perhaps even with text lines in between). But I guess that is a separate issue already.

**Comment by wollmers** [comment]
@bertsky Interesting. The false positive glyphs in the mask are all connected to the horizontal separators. This does not look like a binarisation problem or "overinking". Looks more like caused by morphological transformation that connects glyphs horizontal and vertical too much. Maybe an internal parameter.

See the rectangular shape of the connections:

![Bildschirmfoto 2021-09-24 um 20 49 13](https://user-images.githubusercontent.com/1275557/134726574-1ba02860-f9c9-411d-9f60-36678b5554cb.png)

**Comment by stweil** [comment]
> So I would argue the default should be changed.

I agree, see the comment above which I wrote two years ago. We could not do that for Tesseract 4 for compatibility reasons, but we are free to do it now for Tesseract 5.

**Comment by bertsky** [comment]
> I agree, see the comment above which I wrote two years ago. We could not do that for Tesseract 4 for compatibility reasons, but we are free to do it now for Tesseract 5.

Splendid. Let's do it then! Is there any documentation regarding staff detection which would neet to be changed? 

(Also, the [additional aspects](https://github.com/tesseract-ocr/tesseract/issues/1255#issuecomment-926075821) of how to provide a useful API for the music regions, and fix the above seen false positives, should be tracked in a separate issue...)
